### PR TITLE
Convert to TypeScript, add support for sponsor pools

### PR DESCRIPTION
**Pull request by dstaley**
This PR converts the script to TypeScript to add some extra reliability, in addition to adding support for sponsor coin pools.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,324 @@
+require("dotenv").config();
+const fs = require("fs").promises;
+import { Award, Config, Finding, Slice, Total } from "./types";
+import { getFindingByRisk, riskSet } from "./shared";
+
+const getPieSlices = (jugdedFindings: Finding[]) => {
```
Comment (adamavenir) [inline index.ts]: lil typo here

Comment (dstaley) [reply index.ts]: 🤦‍♀️

**Comment by dstaley** [comment]
@ninek9 what do you think about me updating this to always look for a `contest-config.json` and `contest-judged.json` file? That way the person running it never has to make changes to a `.js` file?

**Comment by ninek9** [comment]
That’d be fantastic

**Pull request merged**